### ISSUE TITLE
No sound output with MOTU 2408 MK3

### ISSUE THREAD

**Issue by Armageddon78** [mdias/rs_asio#24]
Hello !

I'm trying to run Rocksmith 2014 remastered with your project but with no luck so far...
I'm using a MOTU 2408 MK3 ASIO device, which works flawlessly with multiple studio softwares on my PC.
The issue I'm having is I get not sound output.
I tried different buffer size, either through my ASIO control pannel or forced in the ini file.
I tried different sample rate.
No result.

I also have a message when Rocksmith has finished launching saying there are two Rocksmith cable connected. I did work around this commenting one of the two input section of your ini file. Why are there two inputs by default ?

Here attache the log file.

Thanks for you help, and thanks for that your effort bringing ASIO to that software !

[RS_ASIO-log.txt](https://github.com/mdias/rs_asio/files/4141929/RS_ASIO-log.txt)

**Comment by mdias** [comment]
The log looks pretty good, and shows asio streaming data.

Perhaps you're expecting the sound to come through other outputs than the first two? Let me know if that's the case.

If you still can't get any audio, can you also post your Rocksmith.ini and RS-ASIO.ini?

**Comment by Armageddon78** [comment]
I'm using outputs 1&2 which are the mains out. Also my audio interface is a rack unit with meters on it and I can't see any meter moving on any output when Rocksmith is running. So I'm quite sure there is no sound output on any of the 8 analog outs.
To me there are no other application using the device while I start Rocksmith. The interface is freely available to any other ASIO application I have and also request exclusivity. So I guess this is fine.

Here are the two log files you requested.
[inifiles.zip](https://github.com/mdias/rs_asio/files/4142393/inifiles.zip)

And another instance of your log file...
[RS_ASIO-log.txt](https://github.com/mdias/rs_asio/files/4142408/RS_ASIO-log.txt)

**Comment by Armageddon78** [comment]
I also have an "audiodump" log file that Rocksmith generates. Maybe it can help...
[audiodump.txt](https://github.com/mdias/rs_asio/files/4142414/audiodump.txt)

**Comment by Armageddon78** [comment]
In case you ask, here attached the settings of my audio interface...
![MOTU Audio Console](https://user-images.githubusercontent.com/60528632/73590634-0298f680-44e5-11ea-8369-36bd7029e10c.jpg)

**Comment by mdias** [comment]
Replace the following configs in your Rocksmith.ini:
```
EnableMicrophone=0
LatencyBuffer=2
ForceDefaultPlaybackDevice=MOTU PCI ASIO
```

There's some cases that apparently need `ForceDefaultPlaybackDevice` for some reason, yours might be one of those. It could also be the `LatencyBuffer`.

Other than that I see some weird mentions of `samplerate 44100` in your audio dump, which should definitely be 48000.
I would suggest trying to get only output to work first (disable the inputs) and only after that gets going, re-enable the outputs. You can do that by commenting out the "Driver=" property on the inputs of the ini file, or just leaving them blank.

**Comment by Armageddon78** [comment]
Hello,
Thanks for your help.
I applied the changes you recommended to the RS ini file, and commented the inputs in the RS_ASIO ini file. But still, I do not get any sound.
Bellow is the new log file.
[RS_ASIO-log.txt](https://github.com/mdias/rs_asio/files/4144677/RS_ASIO-log.txt)

**Comment by Armageddon78** [comment]
Also, just to let you know : I get sound when I use the ASIO4ALL driver mapped to my MOTU audio outputs. But the sound is ugly as hell (distorted and somewhat slowed down).
The log file : 
[RS_ASIO-log.txt](https://github.com/mdias/rs_asio/files/4144783/RS_ASIO-log.txt)

**Comment by mdias** [comment]
So, the thing that is suspicious about your log withthe MOTU is that on the channel enumeration, for some reason you have "isActive: 1" on the channels. I don't remember seeing that, and I'm not sure if this means ASIO is already being used by another software..?

As for ASIO4ALL, we can actually see "isActive: 0" on that one, which is what I would expect. The reason you get distorted audio there is probably because you have too high of a buffer size. Run the game in windowed mode (so that you can access ASIO4ALL tray icon) and set ASIO4ALL to use 128 too, and then restart the game.

I wonder if there's anything I could do regarding the MOTU drivers... I'll let you know if there's something you can try other than making sure nothing else is using the interface.

**Comment by Armageddon78** [comment]
Well, at the time I launch Rocksmith, there is no other software there that can be using the interface.
The only thing I'm thinking of is the fact the device inputs & outputs are available through Asio and Windows. Is it possible the fact Windows also has access to the i/o makes it active?
Nethertheless this does not block any other software and I'm using a few.

**Comment by mdias** [comment]
In an attempt to try to figure out what is going on, I've made 2 test-builds dor you to try.
Please be advises that these builds will create **very loud noise on all your interface's outputs** so please **turn any speakers volumes down**, or somehow lower the output's volume to avoid an unpleasent surprise to your ears.

- [release-0.3.11-noise-out.zip](https://drive.google.com/open?id=1omJTPVDA7kce_dWIF2GtdBUVxnKwzJwF)
- [release-0.3.11-noise-out-no-timeinfo.zip](https://drive.google.com/open?id=1nxhxI4RvR1xXszQJiU2x-6U7ltEIjMJr)

Note that these zip files only include RS_ASIO.dll, but you still need the other DLL and config file as seen on a normal release.

Please tell me if any, or both of these builds are able to output noise on your outputs, and also attach the log for one of them (only need 1 log).

**Comment by Armageddon78** [comment]
Thanks for your tenacity, this issue does not seem easy to figure out ! :)
So I tried your new DLLs and here are the results : 
release-0.3.11-noise-out.zip > did nothing (not a sound)
release-0.3.11-noise-out-no-timeinfo.zip > same result (not a sound)
Attached are the two log files.
[RS_ASIO-log1.txt](https://github.com/mdias/rs_asio/files/4150596/RS_ASIO-log1.txt)
[RS_ASIO-log2.txt](https://github.com/mdias/rs_asio/files/4150597/RS_ASIO-log2.txt)

I'm starting to think I will be stuck using that poor Rocksmith USB cable...

**Comment by mdias** [comment]
To be honest I have no idea what could be wrong there. Log seems fine and indicates the audio-loop is good but you get no sound, and that weird "isActive: 0" remains...

Now, ofc there's another way, but it may be annoying for you to use.

You can use Voicemeeter which will create a virtual ASIO device that redirects (and if you want can also mix other audio etc) between that virtual device and a true ASIO device. Someone measured the increase in latency to be about 1ms, so it's negligable.

After installing it, you'll have available an ASIO device called "Voicemeeter Virtual ASIO" which you should test in some ASIO app (because it's easier than opening and closing the game to check) and when you get the guitar input channel patched into the virtual ASIO device, your applications should be able to get the signal.

When you get this working, you can configure RS_ASIO to use this ASIO device too, which is known to work fine (I also tried it and it works here).

The inconvenience of this method is that now everytime you want to play Rocksmith, you'll have to open voicemeeter first (it remembers the last settings, so that's good) and then launch the game. But it tends to be a good solution for cases where for some reason RS_ASIO can't interface with an ASIO interface directly for some reason.

Don't forget to restore the normal release RS_ASIO.dll file though :)

**Comment by Armageddon78** [comment]
Very interesting ! This sound like a routing solution. With Voicemeeter will it be possible to send the guitar input to a daw loaded with a good amp sim while playing the game ?

**Comment by mdias** [comment]
Indeed that should be the case.

You should go for [Voicemeeter Potato](https://www.vb-audio.com/Voicemeeter/potato.htm) if you want maximum flexibility.

**Comment by mdias** [comment]
There is now [this new preview release](https://github.com/mdias/rs_asio/releases/tag/v0.4.0) which implements some new ASIO stuff that may make your interface work. These calls should be optional according to the ASIO documentation but apparently some drivers don't work correctly without it, so maybe that's your case?

**Comment by Armageddon78** [comment]
Hey ! You did it ! It works now !
Congratulations. It seems those additional calls were the missing thing to make it work on my MOTU.
Thank you so much !


----- Mail original -----
De: "Micael Dias" <notifications@github.com>
À: "mdias/rs_asio" <rs_asio@noreply.github.com>
Cc: "Armageddon78" <armageddon@free.Fr>, "Author" <author@noreply.github.com>
Envoyé: Jeudi 13 Février 2020 02:42:15
Objet: Re: [mdias/rs_asio] No sound output with MOTU 2408 MK3 (#24)



There is now this new preview release which implements some new ASIO stuff that may make your interface work. These calls should be optional according to the ASIO documentation but apparently some drivers don't work correctly without it, so maybe that's your case?

**Comment by Armageddon78** [comment]
Issue solved with release v0.4.0
